test/tests/functional/pbs_qstat.py

```diff
@@ -94,22 +99,27 @@ def validate_el(self, attr_dict, user):
             qstat_o['restrict_res_to_release_on_suspend'] = 'ncpus'
 
         for each in attr_dict:
-            if each in qstat_o:
+            self.assertIn(each, qstat_o, "Unexpected attribute \"%s\""
+                          " encountered in output of qstat -Bf" %each)
+            if qstat_o[each] != None:
                 if "state_count" in each:
                     out1 = []
                     out = attr_dict[each].split()
                     for k in out:
                         out1.append(k.split(':')[0])
                     self.assertListEqual(qstat_o[each], out1,
                                          "Value of server attribute \"%s\""
-                                         "does not match expected"
-                                         "value" % each)
+                                         " does not match expected"
+                                         " value" % each)
                 else:
                     self.assertEqual(qstat_o[each], attr_dict[each],
                                      "Value of server attribute \"%s\""
-                                     "does not match expected value" % each)
-            self.logger.info("Server attribute \"%s\""
-                             " is validated successfully" % each)
+                                     " does not match expected value" % each)
+                self.logger.info("Server attribute \"%s\""
+                                 " is validated successfully" % each)
+        value = { k : qstat_o[k] for k in set(qstat_o) - set(attr_dict) }
+        self.assertEqual(len(value), 0, "Output of qstat -Bf is missing"
+                                        " some expected attributes - %s" % value.keys())
```

Comment:
Since you only want the keys from the value dictionary, you could just create a list instead of a dictionary with your list comprehension.